src/jsonpatcherproxy.js

```diff
@@ -143,7 +144,9 @@ var JSONPatcherProxy = (function() {
         return Reflect.deleteProperty(target, key);
       }
     });
-    return proxy;
+    /* keeping track of all the proxies to be able to revoke them later */
+    this.proxifiedObjectsMap.push(proxy);
+    return proxy.proxy;
   };
   //grab tree's leaves one by one, encapsulate them into a proxy and return
   JSONPatcherProxy.prototype._proxifyObjectTreeRecursively = function(
```

Comment:
@alshakero  The jsbin throws 504 to me :| could you send it once again?

Forgive me if I get something wrong. I still don't get why can we `revoke` on `deleteProperty` by default.

> But the observed object in JSONPatcherProxies is not ours and I guess we should leave it, and its children, usable.

If child **proxy** object "was cached earlier on purpose for later usages". It's a proxy object, so it's ours and we can destroy it. If user wants to cache his/hers object for later usages after it's unobserved by us, he/she can use `originalObject` or the object returned by delete/splice - I believe it's the original one https://github.com/Palindrom/JSONPatcherProxy/pull/4/files?w=1#diff-3bfa726203c6cb018f2321c18ca9298eR94 . I'm not sure, @alshakero is it?